pyvo/dal/sia2.py

```diff
@@ -221,7 +219,7 @@ def search(self, pos=None, band=None, time=None, pol=None,
                          instrument=instrument, data_type=data_type,
                          calib_level=calib_level, target_name=target_name,
                          res_format=res_format, maxrec=maxrec,
-                         session=session, **kwargs).execute()
+                         session=self._session, **kwargs).execute()
```

Comment:
I don't know if this concern is a big deal, but if someone right now sends passes their own session when calling the `search` method, that would be silently ignored. The alternative is to look for the session into `kwargs` and use that one if available. At least temporarily.

Reply:
Shouldn't they have used that session in their instance init? (as far as I see we basically pass on `self._session` at this point in the other classes, too (though sometimes it happens in the `create_query` method, rather than this high up.)

Reply:
I think this is an interesting question.  I understand the concern but feel like breaking someone's existing behavior is unlikely.  If we find this is a problem moving forward, we can come back and address it again.  The use of `kwargs` throughout this code makes tracking some of the parameters challenging.

Reply:
I agree about `kwargs`. If I'm not mistaken, these arguments are intended primarily for the `requests` underlying library so it's easy for someone to specify things like `timeout`, `headers`, etc with call. Maybe that's too confusing and we need to add a request attribute map in the service class for that more expert-use purpose. 

The bottom line is that we are essentially removing `sessions` from the list of arguments, silently without giving any warning. There might not be any users of this feature, but if there are, they will have a very hard time understanding what is going on.

Reply:
I've tried to trace some of these kwagrs, and they ended up being an rendered out to be attributes, most of not being used at all. So, if the aim is to pass them along all the way to `request`, then I would suggest to add a `request_kwargs` parameter that accepts an dict. 

But changing it all at this point would potentially painful for users (though I would say any errors would be easy to fix, or in fact directly highlighting bugs in their usage.

Reply:
I'm OK with `request_kwargs`. And a deprecation warning when `kwargs` is not empty? A pain, I know, but the right etiquette.

Reply:
I'll check, even the deprecation decorators may work, I just never tried them for `**` cases.

Reply:
I feel I'm just a broken record with the same ideas (#466) 😅